app/javascript/reducers/greetingReducer.js

```diff
@@ -0,0 +1,12 @@
+const initialState = '';
+
+const greetingReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case 'SET_RANDOM_GREETING':
+      return action.payload;
+    default:
+      return state;
+  }
+};
+
+export default greetingReducer;
```

Comment:
- [ ] As mentioned by the previous code reviewer, you should implement the `redux` thunk properly, It seems like you have a lot of issues in the `Redux` part, which is a very essential aspect of front-end application, which helps to send or get, to or from the database through `Back-end` API, Kindly fix the issues, that are difficult to debug, because you the build file production settings file. That's why please focus on fixing your `Redux` part, you can take the example of `React Group Project`, to solve the issue, just as a reference for implementing the redux properly.